src/Microsoft.Health.Core/Features/HealthPublisher/HealthCheckMetricPublisher.cs

```diff
@@ -0,0 +1,53 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Core.Features.Health;
+using Microsoft.Health.Core.Features.Metric;
+
+namespace Microsoft.Health.Core.Features.HealthPublisher;
+internal class HealthCheckMetricPublisher : IHealthCheckPublisher
+{
+    private readonly IResourceHealthSignalProvider _resourceHealthMeter;
```

Comment:
Yeah I did rename this actually to `_resourceHealthSignalProvider` in the last commit. I originally had this using the meter naming because I was imagining it just as a plain meter, but I updated it to this "singal provider" since the Meter doesnt typically handle emitting the metric / adding the dimensions itself. And to keep it more generic since fhir is using this as well.
So in our implementation, we will have a DicomResourceHealthSignalProvider that has a method to add the dimensions and emit this metric, and that class will have a separate Meter